R/bootstrap.nlsfit.R

```diff
@@ -379,10 +397,19 @@ bootstrap.nlsfit <- function(fn,
   }else{
     ## the format of gr has to be nrows=length(par), ncols=length(Y)
     if(errormodel == "yerrors"){
+      grpriors <- c()
+      if(is.null(priors$param)==FALSE && is.null(priors$p)==FALSE && is.null(priors$psamples)==FALSE){
+        npriors <- length(priors$param)
+        npar <- length(par.guess)
+        for (i in 1:npriors) {
+          aux <- t(c(rep(0, (priors$param[i] - 1)), 1, rep(0, (npar - priors$param[i]))))
+          grpriors <- rbind(grpriors, aux)
+        }
```

Comment:
From looking at the code I believe that this is correct. Here's my reasoning:

This is the description of the `gr` function:

```r
#' @param gr `gr(par, x, ...)`. \code{gr=d(fn) / d(par)} is a function to
#' return the gradient of \code{fn}. It must return an array with
#' return the gradient of \code{fn}. It must return an array with
#' \code{length(x)} rows and \code{length(par)} columns.
#' \code{length(x)} rows and \code{length(par)} columns.
```

It gives a matrix with has as many rows as there is dependent variables (`y`) and as many columns as there are parameters. We are not adding any more parameters but more dependent variables in forms of priors. Therefore we need to extend it with more rows, the `rbind` is exactly right.

The derivative of the additional priors with respect to the parameters should just be a 1 at the column corresponding to the parameter.

Although in general I favor construction of something with a single expression and not altering it, I thought that *maybe* it would be easier to read like this:

```r
aux <- rep(0, npar)
aux[priors$param[i]] <- 1
```

Reply:
This seems to be a nice simplification so I just made those changes.